podium-gateway/src/main/java/nl/thehyve/podium/web/rest/RequestResource.java

```diff
@@ -605,4 +617,89 @@
+                                          @RequestParam("file") MultipartFile file) throws URISyntaxException,
+                                                                                           ActionNotAllowed,
+                                                                                           IOException{
+        AuthenticatedUser user = securityService.getCurrentUser();
+
+        if(!file.isEmpty()){
+            RequestFileRepresentation requestFileRepresentation = requestFileService.addFile(user, uuid, file,
+                RequestFileType.NONE);
+            return ResponseEntity.accepted().body(requestFileRepresentation);
+
+        } else {
+            throw new IOException("File empty");
+        }
+    }
+
+    /**
+     * Return a file for a given request
+     * @return the requested file
+     */
+    @GetMapping("/requests/{uuid}/files/{fileuuid}")
+    @SecuredByAuthority({AuthorityConstants.ORGANISATION_ADMIN, AuthorityConstants.ORGANISATION_COORDINATOR,
+                         AuthorityConstants.REVIEWER, AuthorityConstants.RESEARCHER})
+    @Timed
+    public ResponseEntity<Object> getFile(@RequestUuidParameter @PathVariable("uuid") UUID requestUuid,
+                                          @PathVariable("fileuuid") UUID fileUuid) throws URISyntaxException,
+        ActionNotAllowed, IOException{
+
+        AuthenticatedUser user = securityService.getCurrentUser();
+        ByteArrayResource resource = requestFileService.getFile(user, fileUuid);
+        return ResponseEntity.ok()
+            .contentType(MediaType.parseMediaType("application/octet-stream"))
+            .body(resource);
+    }
+
+    @GetMapping("/requests/{uuid}/files")
+    @SecuredByAuthority({AuthorityConstants.ORGANISATION_ADMIN, AuthorityConstants.ORGANISATION_COORDINATOR,
+                         AuthorityConstants.REVIEWER, AuthorityConstants.RESEARCHER})
+    @Timed
+    public ResponseEntity<List<RequestFileRepresentation>> listFile(@RequestUuidParameter @PathVariable("uuid") UUID requestUuid) throws
+        URISyntaxException, ActionNotAllowed {
+        AuthenticatedUser user = securityService.getCurrentUser();
+
+        List<RequestFileRepresentation> files = requestFileService.getFilesForRequest(user, requestUuid);
+
+        return ResponseEntity.ok(files);
+    }
+
+    @DeleteMapping("/requests/deletefile/{fileuuid}")
+    @SecuredByAuthority({AuthorityConstants.ORGANISATION_ADMIN, AuthorityConstants.ORGANISATION_COORDINATOR,
+                         AuthorityConstants.REVIEWER, AuthorityConstants.RESEARCHER})
+    @Timed
+    public ResponseEntity<Object> deleteFile(@PathVariable("fileuuid") UUID fileUuid)
+        throws URISyntaxException, ActionNotAllowed, IOException{
+
+        AuthenticatedUser user = securityService.getCurrentUser();
+
+        requestFileService.deleteFile(user, fileUuid);
+
+        return ResponseEntity.ok("ok");
```

Comment:
How about return type `ResponseEntity<Void>` and not returning anything?

Reply:
Riza's branch has an alternative where we copied what we do when request drafts are created see: https://github.com/thehyve/podium/pull/237/files